stacks-signer/src/stacks_client.rs

```diff
@@ -320,6 +403,46 @@ impl StacksClient {
             .to_string();
         Ok(result)
     }
+
+    fn pox_path(&self) -> String {
+        format!("{}/v2/pox", self.http_origin)
+    }
+
+    fn transaction_path(&self) -> String {
+        format!("{}/v2/transactions", self.http_origin)
+    }
+
+    fn read_only_path(
+        &self,
+        contract_addr: &StacksAddress,
+        contract_name: &ContractName,
+        function_name: &ClarityName,
+    ) -> String {
+        format!(
+            "{}/v2/contracts/call-read/{contract_addr}/{contract_name}/{function_name}",
+            self.http_origin
+        )
+    }
+}
+
+/// Retry a function F with an exponential backoff and notification on transient failure
+pub fn retry_with_exponential_backoff<F, E, T>(request_fn: F) -> Result<T, ClientError>
```

Comment:
I'd be curious as to how often you need to retry requests?  The blockchain should strive to get this down to zero.

Reply:
We ran into a problem where hammering `stacks-node` during startup caused it to go into a bad state.  We didn't try to diagnose the problem, though we should have filed a ticket.  Whatever the state was it prevented the node from bootstrapping.  Adding large retry timeouts (O(seconds)) fixed it.   There were retry timeouts already in some places in the code, but now they cover all cases not just connection failures.

During normal operation we didn't usually see failures.

Reply:
Unfortunately...often . Bceause in the integration test. The node will not start up if the signer isn't up. But the signer cannot be up without trying to contact the node. :P So we initialize the signer and open its listener port. The node starts up and connects to it. But immediately the run_one_loop starts running and starts hammering the node which is still starting up...and this causes it to hang. So the backoff keeps it from destroying it as it tries to connect for the necessary info (DKG key for example)